{
  "cells": [
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "cff35ae5",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 478
        },
        "id": "cff35ae5",
        "outputId": "46b7894f-34a5-4b99-967b-923b757497cf"
      },
      "outputs": [
        {
          "output_type": "error",
          "ename": "KeyError",
          "evalue": "'Gene'",
          "traceback": [
            "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
            "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3652\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3653\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcasted_key\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3654\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/pandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/pandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
            "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
            "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
            "\u001b[0;31mKeyError\u001b[0m: 'Gene'",
            "\nThe above exception was the direct cause of the following exception:\n",
            "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
            "\u001b[0;32m<ipython-input-9-8de2e36a3f19>\u001b[0m in \u001b[0;36m<cell line: 12>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     10\u001b[0m \u001b[0;31m# Convert chemical names to lowercase for both DataFrames\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     11\u001b[0m \u001b[0mdf1\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Chemicals'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdf1\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Chemicals'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlower\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 12\u001b[0;31m \u001b[0mdf2\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Gene'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdf2\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Gene'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlower\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     13\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     14\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3759\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnlevels\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3760\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_multilevel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3761\u001b[0;31m             \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3762\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mis_integer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3763\u001b[0m                 \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;32m/usr/local/lib/python3.10/dist-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3653\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcasted_key\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3654\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3655\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3656\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mTypeError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3657\u001b[0m             \u001b[0;31m# If we have a listlike key, _check_indexing_error will raise\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
            "\u001b[0;31mKeyError\u001b[0m: 'Gene'"
          ]
        }
      ],
      "source": [
        "import pandas as pd\n",
        "\n",
        "# Read the first CSV file\n",
        "df1 = pd.read_csv('Task4_secondary_metabolites_fungus_Chemicals_pubmed.csv')\n",
        "\n",
        "# Read the second CSV file\n",
        "df2 = pd.read_csv('/content/Task4_Alkaline_gene_pubmed (1).csv')\n",
        "\n",
        "\n",
        "# Convert chemical names to lowercase for both DataFrames\n",
        "df1['Chemicals'] = df1['Chemicals'].str.lower()\n",
        "df2['Gene'] = df2['Gene'].str.lower()\n",
        "\n",
        "\n",
        "# # Merge the two DataFrames on 'PMID' column, ensuring chemical names are not the same\n",
        "merged_df = pd.merge(df1, df2, on='PMID', how='inner', validate='m:m')\n",
        "\n",
        "merged_df\n",
        "\n",
        "# # Save the merged DataFrame to a new CSV file\n",
        "# merged_df.to_csv('gene_chemical_data.csv', index=False)\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "41543190",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "41543190",
        "outputId": "15ff3c7a-eb94-4e66-8a7c-de82948452e7"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "             Chemicals ChemicalsID      PMID   PMCID_x         Gene  GeneID  \\\n",
              "0                water     D014867  32533304   7347518          tap   10482   \n",
              "1                 tmao     C005855  36916005  10111349          prb   10536   \n",
              "9       fos-choline-12     D002794  37577439  10416648        ndh-2    1660   \n",
              "10                nadh     D009243  37577439  10416648        ndh-2    1660   \n",
              "11             glycine     D005998  37577439  10416648        ndh-2    1660   \n",
              "12              saline     D012965  34256694   8278772         pip1  856754   \n",
              "13               water     D014867  34256694   8278772         pip1  856754   \n",
              "14        carbohydrate     D002241  34093984   8148631          tak    1025   \n",
              "15            hydrogen     D006859  34093984   8148631          tak    1025   \n",
              "16               lipid     D008055  27276261   4906265           ph    5053   \n",
              "17           hydroxide     C031356  27276261   4906265           ph    5053   \n",
              "18       phospholipids     D010743  27276261   4906265           ph    5053   \n",
              "19               lipid     D008055  27276261   4906265           ph    5053   \n",
              "20         valinomycin     D014634  27276261   4906265           ph    5053   \n",
              "21             carbons     D002244  27276261   4906265           ph    5053   \n",
              "22              lipids     D008055  27276261   4906265           ph    5053   \n",
              "23          isoprenoid     D013729  27276261   4906265           ph    5053   \n",
              "24               water     D014867  27276261   4906265           ph    5053   \n",
              "25               water     D014867  28045976   5207672          l-1   16728   \n",
              "26               water     D014867  28045976   5207672         ml-1   16728   \n",
              "27              oxygen     D010100  28045976   5207672          l-1   16728   \n",
              "28              oxygen     D010100  28045976   5207672         ml-1   16728   \n",
              "29               water     D014867  24927538   4156692          fog  161882   \n",
              "30            nitrogen     D009584  24927538   4156692          fog  161882   \n",
              "31                  o2     D010100  24927538   4156692          fog  161882   \n",
              "32              copper     D003300  22295871   3330824    caspase-1     834   \n",
              "33              copper     D003300  22295871   3330824    caspase-1     834   \n",
              "35               water     D014867  22295871   3330824    caspase-1     834   \n",
              "36               water     D014867  22295871   3330824    caspase-1     834   \n",
              "38         vermistatin     C051060  22295871   3330824    caspase-1     834   \n",
              "39         vermistatin     C051060  22295871   3330824    caspase-1     834   \n",
              "41                 eap     C005448  30400922   6220464  osteocalcin   12097   \n",
              "43             calcium     D002118  30400922   6220464  osteocalcin   12097   \n",
              "44  risedronate sodium  D000068296  30400922   6220464  osteocalcin   12097   \n",
              "45                 res  D000068296  30400922   6220464  osteocalcin   12097   \n",
              "\n",
              "     PMCID_y                                          Sentences  \n",
              "0    7347518   While many studies have investigated microorg...  \n",
              "1   10111349                                                NaN  \n",
              "9   10416648   Here, we performed a combination of whole cel...  \n",
              "10  10416648   Within this data set we revealed a full oxida...  \n",
              "11  10416648   We also observed C. thermarum TA2.A1 expressi...  \n",
              "12   8278772  Our results indicate that AQPs play an importa...  \n",
              "13   8278772   Seawater and drought tolerance in this specie...  \n",
              "14   8148631   Addition of a carbohydrate binding\\ndomain (C...  \n",
              "15   8148631   Xyn11 structure, which has been solved at\\n1....  \n",
              "16   4906265  Extremophile archaea-inspired lipid structures...  \n",
              "17   4906265  This work explores the proton/hydroxide permea...  \n",
              "18   4906265  This work explores the proton/hydroxide permea...  \n",
              "19   4906265  This work explores the proton/hydroxide permea...  \n",
              "20   4906265   All permeability assays were performed in the...  \n",
              "21   4906265   Among tethered lipid monolayers, PH+/OH− incr...  \n",
              "22   4906265  This work explores the proton/hydroxide permea...  \n",
              "23   4906265   A fluorescence-based permeability assay was o...  \n",
              "24   4906265   Molecular dynamics simulations revealed a str...  \n",
              "25   5207672  This study describes, for the first time, the ...  \n",
              "26   5207672  This study describes, for the first time, the ...  \n",
              "27   5207672   In situ daily photosynthetic oxygen productio...  \n",
              "28   5207672   In situ daily photosynthetic oxygen productio...  \n",
              "29   4156692   Liquid water is the common ecological require...  \n",
              "30   4156692   Some nitrogen must be present for life and th...  \n",
              "31   4156692   Levels of O2 over a few percent on an exoplan...  \n",
              "32   3330824  Berkeley Pit Lake, Butte, Montana is a 540 m. ...  \n",
              "33   3330824  Berkeley Pit Lake, Butte, Montana is a 540 m. ...  \n",
              "35   3330824  Berkeley Pit Lake, Butte, Montana is a 540 m. ...  \n",
              "36   3330824  Berkeley Pit Lake, Butte, Montana is a 540 m. ...  \n",
              "38   3330824   Several polyketide metabolites including the ...  \n",
              "39   3330824   Several polyketide metabolites including the ...  \n",
              "41   6220464  Extracellular polymeric substances isolated fr...  \n",
              "43   6220464   Variations in body weight gains during the ex...  \n",
              "44   6220464   The results for the mixed formulations of EAP...  \n",
              "45   6220464   The results for the mixed formulations of EAP...  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-22b5df28-4a50-419c-b088-b7f2c515c2d0\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Chemicals</th>\n",
              "      <th>ChemicalsID</th>\n",
              "      <th>PMID</th>\n",
              "      <th>PMCID_x</th>\n",
              "      <th>Gene</th>\n",
              "      <th>GeneID</th>\n",
              "      <th>PMCID_y</th>\n",
              "      <th>Sentences</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>water</td>\n",
              "      <td>D014867</td>\n",
              "      <td>32533304</td>\n",
              "      <td>7347518</td>\n",
              "      <td>tap</td>\n",
              "      <td>10482</td>\n",
              "      <td>7347518</td>\n",
              "      <td>While many studies have investigated microorg...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>tmao</td>\n",
              "      <td>C005855</td>\n",
              "      <td>36916005</td>\n",
              "      <td>10111349</td>\n",
              "      <td>prb</td>\n",
              "      <td>10536</td>\n",
              "      <td>10111349</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>fos-choline-12</td>\n",
              "      <td>D002794</td>\n",
              "      <td>37577439</td>\n",
              "      <td>10416648</td>\n",
              "      <td>ndh-2</td>\n",
              "      <td>1660</td>\n",
              "      <td>10416648</td>\n",
              "      <td>Here, we performed a combination of whole cel...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10</th>\n",
              "      <td>nadh</td>\n",
              "      <td>D009243</td>\n",
              "      <td>37577439</td>\n",
              "      <td>10416648</td>\n",
              "      <td>ndh-2</td>\n",
              "      <td>1660</td>\n",
              "      <td>10416648</td>\n",
              "      <td>Within this data set we revealed a full oxida...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11</th>\n",
              "      <td>glycine</td>\n",
              "      <td>D005998</td>\n",
              "      <td>37577439</td>\n",
              "      <td>10416648</td>\n",
              "      <td>ndh-2</td>\n",
              "      <td>1660</td>\n",
              "      <td>10416648</td>\n",
              "      <td>We also observed C. thermarum TA2.A1 expressi...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12</th>\n",
              "      <td>saline</td>\n",
              "      <td>D012965</td>\n",
              "      <td>34256694</td>\n",
              "      <td>8278772</td>\n",
              "      <td>pip1</td>\n",
              "      <td>856754</td>\n",
              "      <td>8278772</td>\n",
              "      <td>Our results indicate that AQPs play an importa...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>water</td>\n",
              "      <td>D014867</td>\n",
              "      <td>34256694</td>\n",
              "      <td>8278772</td>\n",
              "      <td>pip1</td>\n",
              "      <td>856754</td>\n",
              "      <td>8278772</td>\n",
              "      <td>Seawater and drought tolerance in this specie...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>14</th>\n",
              "      <td>carbohydrate</td>\n",
              "      <td>D002241</td>\n",
              "      <td>34093984</td>\n",
              "      <td>8148631</td>\n",
              "      <td>tak</td>\n",
              "      <td>1025</td>\n",
              "      <td>8148631</td>\n",
              "      <td>Addition of a carbohydrate binding\\ndomain (C...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>15</th>\n",
              "      <td>hydrogen</td>\n",
              "      <td>D006859</td>\n",
              "      <td>34093984</td>\n",
              "      <td>8148631</td>\n",
              "      <td>tak</td>\n",
              "      <td>1025</td>\n",
              "      <td>8148631</td>\n",
              "      <td>Xyn11 structure, which has been solved at\\n1....</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>16</th>\n",
              "      <td>lipid</td>\n",
              "      <td>D008055</td>\n",
              "      <td>27276261</td>\n",
              "      <td>4906265</td>\n",
              "      <td>ph</td>\n",
              "      <td>5053</td>\n",
              "      <td>4906265</td>\n",
              "      <td>Extremophile archaea-inspired lipid structures...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>17</th>\n",
              "      <td>hydroxide</td>\n",
              "      <td>C031356</td>\n",
              "      <td>27276261</td>\n",
              "      <td>4906265</td>\n",
              "      <td>ph</td>\n",
              "      <td>5053</td>\n",
              "      <td>4906265</td>\n",
              "      <td>This work explores the proton/hydroxide permea...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>18</th>\n",
              "      <td>phospholipids</td>\n",
              "      <td>D010743</td>\n",
              "      <td>27276261</td>\n",
              "      <td>4906265</td>\n",
              "      <td>ph</td>\n",
              "      <td>5053</td>\n",
              "      <td>4906265</td>\n",
              "      <td>This work explores the proton/hydroxide permea...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>19</th>\n",
              "      <td>lipid</td>\n",
              "      <td>D008055</td>\n",
              "      <td>27276261</td>\n",
              "      <td>4906265</td>\n",
              "      <td>ph</td>\n",
              "      <td>5053</td>\n",
              "      <td>4906265</td>\n",
              "      <td>This work explores the proton/hydroxide permea...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>20</th>\n",
              "      <td>valinomycin</td>\n",
              "      <td>D014634</td>\n",
              "      <td>27276261</td>\n",
              "      <td>4906265</td>\n",
              "      <td>ph</td>\n",
              "      <td>5053</td>\n",
              "      <td>4906265</td>\n",
              "      <td>All permeability assays were performed in the...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>21</th>\n",
              "      <td>carbons</td>\n",
              "      <td>D002244</td>\n",
              "      <td>27276261</td>\n",
              "      <td>4906265</td>\n",
              "      <td>ph</td>\n",
              "      <td>5053</td>\n",
              "      <td>4906265</td>\n",
              "      <td>Among tethered lipid monolayers, PH+/OH− incr...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>22</th>\n",
              "      <td>lipids</td>\n",
              "      <td>D008055</td>\n",
              "      <td>27276261</td>\n",
              "      <td>4906265</td>\n",
              "      <td>ph</td>\n",
              "      <td>5053</td>\n",
              "      <td>4906265</td>\n",
              "      <td>This work explores the proton/hydroxide permea...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>23</th>\n",
              "      <td>isoprenoid</td>\n",
              "      <td>D013729</td>\n",
              "      <td>27276261</td>\n",
              "      <td>4906265</td>\n",
              "      <td>ph</td>\n",
              "      <td>5053</td>\n",
              "      <td>4906265</td>\n",
              "      <td>A fluorescence-based permeability assay was o...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>24</th>\n",
              "      <td>water</td>\n",
              "      <td>D014867</td>\n",
              "      <td>27276261</td>\n",
              "      <td>4906265</td>\n",
              "      <td>ph</td>\n",
              "      <td>5053</td>\n",
              "      <td>4906265</td>\n",
              "      <td>Molecular dynamics simulations revealed a str...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>25</th>\n",
              "      <td>water</td>\n",
              "      <td>D014867</td>\n",
              "      <td>28045976</td>\n",
              "      <td>5207672</td>\n",
              "      <td>l-1</td>\n",
              "      <td>16728</td>\n",
              "      <td>5207672</td>\n",
              "      <td>This study describes, for the first time, the ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>26</th>\n",
              "      <td>water</td>\n",
              "      <td>D014867</td>\n",
              "      <td>28045976</td>\n",
              "      <td>5207672</td>\n",
              "      <td>ml-1</td>\n",
              "      <td>16728</td>\n",
              "      <td>5207672</td>\n",
              "      <td>This study describes, for the first time, the ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>27</th>\n",
              "      <td>oxygen</td>\n",
              "      <td>D010100</td>\n",
              "      <td>28045976</td>\n",
              "      <td>5207672</td>\n",
              "      <td>l-1</td>\n",
              "      <td>16728</td>\n",
              "      <td>5207672</td>\n",
              "      <td>In situ daily photosynthetic oxygen productio...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>28</th>\n",
              "      <td>oxygen</td>\n",
              "      <td>D010100</td>\n",
              "      <td>28045976</td>\n",
              "      <td>5207672</td>\n",
              "      <td>ml-1</td>\n",
              "      <td>16728</td>\n",
              "      <td>5207672</td>\n",
              "      <td>In situ daily photosynthetic oxygen productio...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>29</th>\n",
              "      <td>water</td>\n",
              "      <td>D014867</td>\n",
              "      <td>24927538</td>\n",
              "      <td>4156692</td>\n",
              "      <td>fog</td>\n",
              "      <td>161882</td>\n",
              "      <td>4156692</td>\n",
              "      <td>Liquid water is the common ecological require...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>30</th>\n",
              "      <td>nitrogen</td>\n",
              "      <td>D009584</td>\n",
              "      <td>24927538</td>\n",
              "      <td>4156692</td>\n",
              "      <td>fog</td>\n",
              "      <td>161882</td>\n",
              "      <td>4156692</td>\n",
              "      <td>Some nitrogen must be present for life and th...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>31</th>\n",
              "      <td>o2</td>\n",
              "      <td>D010100</td>\n",
              "      <td>24927538</td>\n",
              "      <td>4156692</td>\n",
              "      <td>fog</td>\n",
              "      <td>161882</td>\n",
              "      <td>4156692</td>\n",
              "      <td>Levels of O2 over a few percent on an exoplan...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>32</th>\n",
              "      <td>copper</td>\n",
              "      <td>D003300</td>\n",
              "      <td>22295871</td>\n",
              "      <td>3330824</td>\n",
              "      <td>caspase-1</td>\n",
              "      <td>834</td>\n",
              "      <td>3330824</td>\n",
              "      <td>Berkeley Pit Lake, Butte, Montana is a 540 m. ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>33</th>\n",
              "      <td>copper</td>\n",
              "      <td>D003300</td>\n",
              "      <td>22295871</td>\n",
              "      <td>3330824</td>\n",
              "      <td>caspase-1</td>\n",
              "      <td>834</td>\n",
              "      <td>3330824</td>\n",
              "      <td>Berkeley Pit Lake, Butte, Montana is a 540 m. ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>35</th>\n",
              "      <td>water</td>\n",
              "      <td>D014867</td>\n",
              "      <td>22295871</td>\n",
              "      <td>3330824</td>\n",
              "      <td>caspase-1</td>\n",
              "      <td>834</td>\n",
              "      <td>3330824</td>\n",
              "      <td>Berkeley Pit Lake, Butte, Montana is a 540 m. ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>36</th>\n",
              "      <td>water</td>\n",
              "      <td>D014867</td>\n",
              "      <td>22295871</td>\n",
              "      <td>3330824</td>\n",
              "      <td>caspase-1</td>\n",
              "      <td>834</td>\n",
              "      <td>3330824</td>\n",
              "      <td>Berkeley Pit Lake, Butte, Montana is a 540 m. ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>38</th>\n",
              "      <td>vermistatin</td>\n",
              "      <td>C051060</td>\n",
              "      <td>22295871</td>\n",
              "      <td>3330824</td>\n",
              "      <td>caspase-1</td>\n",
              "      <td>834</td>\n",
              "      <td>3330824</td>\n",
              "      <td>Several polyketide metabolites including the ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>39</th>\n",
              "      <td>vermistatin</td>\n",
              "      <td>C051060</td>\n",
              "      <td>22295871</td>\n",
              "      <td>3330824</td>\n",
              "      <td>caspase-1</td>\n",
              "      <td>834</td>\n",
              "      <td>3330824</td>\n",
              "      <td>Several polyketide metabolites including the ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>41</th>\n",
              "      <td>eap</td>\n",
              "      <td>C005448</td>\n",
              "      <td>30400922</td>\n",
              "      <td>6220464</td>\n",
              "      <td>osteocalcin</td>\n",
              "      <td>12097</td>\n",
              "      <td>6220464</td>\n",
              "      <td>Extracellular polymeric substances isolated fr...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>43</th>\n",
              "      <td>calcium</td>\n",
              "      <td>D002118</td>\n",
              "      <td>30400922</td>\n",
              "      <td>6220464</td>\n",
              "      <td>osteocalcin</td>\n",
              "      <td>12097</td>\n",
              "      <td>6220464</td>\n",
              "      <td>Variations in body weight gains during the ex...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>44</th>\n",
              "      <td>risedronate sodium</td>\n",
              "      <td>D000068296</td>\n",
              "      <td>30400922</td>\n",
              "      <td>6220464</td>\n",
              "      <td>osteocalcin</td>\n",
              "      <td>12097</td>\n",
              "      <td>6220464</td>\n",
              "      <td>The results for the mixed formulations of EAP...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>45</th>\n",
              "      <td>res</td>\n",
              "      <td>D000068296</td>\n",
              "      <td>30400922</td>\n",
              "      <td>6220464</td>\n",
              "      <td>osteocalcin</td>\n",
              "      <td>12097</td>\n",
              "      <td>6220464</td>\n",
              "      <td>The results for the mixed formulations of EAP...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-22b5df28-4a50-419c-b088-b7f2c515c2d0')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-22b5df28-4a50-419c-b088-b7f2c515c2d0 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-22b5df28-4a50-419c-b088-b7f2c515c2d0');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-f96027ed-164c-4b42-80b4-aedb44432305\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-f96027ed-164c-4b42-80b4-aedb44432305')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-f96027ed-164c-4b42-80b4-aedb44432305 button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "dataframe",
              "variable_name": "merged_df",
              "summary": "{\n  \"name\": \"merged_df\",\n  \"rows\": 35,\n  \"fields\": [\n    {\n      \"column\": \"Chemicals\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 24,\n        \"samples\": [\n          \"lipid\",\n          \"nitrogen\",\n          \"water\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"ChemicalsID\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 21,\n        \"samples\": [\n          \"D014867\",\n          \"C051060\",\n          \"D009584\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"PMID\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 4723924,\n        \"min\": 22295871,\n        \"max\": 37577439,\n        \"num_unique_values\": 10,\n        \"samples\": [\n          22295871,\n          36916005,\n          27276261\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"PMCID_x\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2184187,\n        \"min\": 3330824,\n        \"max\": 10416648,\n        \"num_unique_values\": 10,\n        \"samples\": [\n          3330824,\n          10111349,\n          4906265\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Gene\",\n      \"properties\": {\n        \"dtype\": \"category\",\n        \"num_unique_values\": 11,\n        \"samples\": [\n          \"ph\",\n          \"tap\",\n          \"caspase-1\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"GeneID\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 201846,\n        \"min\": 834,\n        \"max\": 856754,\n        \"num_unique_values\": 10,\n        \"samples\": [\n          834,\n          10536,\n          5053\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"PMCID_y\",\n      \"properties\": {\n        \"dtype\": \"number\",\n        \"std\": 2184187,\n        \"min\": 3330824,\n        \"max\": 10416648,\n        \"num_unique_values\": 10,\n        \"samples\": [\n          3330824,\n          10111349,\n          4906265\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    },\n    {\n      \"column\": \"Sentences\",\n      \"properties\": {\n        \"dtype\": \"string\",\n        \"num_unique_values\": 34,\n        \"samples\": [\n          \" A fluorescence-based permeability assay was optimized to determine the effects on the PH+/OH\\u2212 through liposome membranes with variations in the following lipid attributes: transmembrane tethering, tether length, and the presence of isoprenoid methyl groups on one or both lipid tails.;  Untethered lipids with two isoprenoid methyl tails led to lower PH+/OH\\u2212 values than lipids with only one or no isoprenoid tails.;  Archaeal lipid tails often contain methyl branches in a repeating isoprenoid pattern, as well as cyclopentane and cyclohexane rings.; To make this study possible, we synthesized a series of\\u00a0extremophile-inspired lipids composed of two glycerophosphocholine heads connected by a membrane-spanning n-alkyl tether with each head ether-bonded to a phytanyl chain (T28, T32, and T36 (Fig.\\u00a01, A\\u2013C)), as well as several\\u00a0untethered lipids (U16, 1,2-di-O-phytanyl-sn-glycero-3-phosphocholine (Di-O-PhyPC), 1,2-diphytanoyl-sn-glycero-3-phosphocholine (DiPhyPC), and POPC (Fig.\\u00a01, D\\u2013G)) to seek answers to our questions about structure-permeability relationships of liposomes from pure lipid preparations at room temperature: First, how does the presence of a single tether between two lipid headgroups affect PH+/OH\\u2212 compared to the absence of a tether? Second, how does the length of the tether affect H+/OH\\u2212 permeability when the lengths of the untethered phytanyl chains are kept constant at 16 carbons? And third, how does the PH+/OH\\u2212 of an untethered lipid with a repeating isoprenoid methyl motif on one of its tails compare to that of a lipid with the methyl groups on both tails?; We explored the effect of transmembrane tethering, tether length, and the presence of isoprenoid methyl groups on one or both lipid tails on the PH+/OH\\u2212 values through liposomes formed from a series of synthetic lipids with small systematic structural variations.;  Given that U16 and Di-O-PhyPC only differ structurally by the presence or absence of isoprenoid methyl groups on one lipid tail, this result agrees with previous findings that this motif lowers H+/OH\\u2212 permeability (8).;  Most lipids characteristic of acidophilic archaea display the isoprenoid methyl motif on both acyl chains (1).;  Further, monopolar lipids with two tails containing isoprenoid methyl branches form membranes with lower H+/OH\\u2212 permeability than those with one branching tail and one unbranched tail.;This work explores the proton/hydroxide permeability (PH+/OH\\u2212) of membranes that were made of synthetic extremophile-inspired phospholipids with systematically varied structural elements.;  A fluorescence-based permeability assay was optimized to determine the effects on the PH+/OH\\u2212 through liposome membranes with variations in the following lipid attributes: transmembrane tethering, tether length, and the presence of isoprenoid methyl groups on one or both lipid tails.;  All permeability assays were performed in the presence of a low concentration of valinomycin (10\\u00a0nM) to prevent buildup of a membrane potential without artificially increasing the measured PH+/OH\\u2212. Surprisingly, the presence of a transmembrane tether did not impact PH+/OH\\u2212 at room temperature.;  Among tethered lipid monolayers, PH+/OH\\u2212 increased with increasing tether length if the number of carbons in the untethered acyl tail was constant.;  Untethered lipids with two isoprenoid methyl tails led to lower PH+/OH\\u2212 values than lipids with only one or no isoprenoid tails.;  Whatever the mechanism, the permeation of protons or hydroxide ions is conventionally described by a permeability coefficient, PH+/OH\\u2212, in units of cm s\\u22121, defined by Fick\\u2019s law (Eq. 1):; where J (mol m\\u22122 s\\u22121) is the inward flux of solute across the membrane, PH+/OH\\u2212 (m s\\u22121) is the permeability of the membrane, and \\u0394C (mol m\\u22123) is the concentration gradient across the membrane.; Widely varying values of PH+/OH\\u2212 have been observed both between and within laboratories.;  Reported PH+/OH\\u2212 values for membranes composed of the lipid 1-palmitoyl-2-oleoyl-sn-glycero-3-phosphocholine (POPC), for example, include values on the order of 10\\u22121 cm s\\u22121 (21), 10\\u22122\\u00a0cm\\u00a0s\\u22121 (22), 10\\u22125 cm s\\u22121 (23), and 10\\u22127 cm s\\u22121 (24, 25); these disparate values were all similarly obtained by subjecting liposomes to a stepwise \\u0394pH and observing proton flux over time.;  Recommendations such as this are sensible; there is a clear need for a standard, well-defined assay for permeation of H+/OH\\u2212 across membranes so that PH+/OH\\u2212 values obtained by different groups can be compared.;  However, Kuyper et\\u00a0al. analyzed H+/OH\\u2212 permeation across hundreds of single vesicles in parallel and observed a high degree of variation in PH+/OH\\u2212 between liposomes in the same experiment without observing correlation with liposome size or any other observable parameter, indicating that broad distributions of PH+/OH\\u2212 values may be unavoidable (24).; In an attempt to obtain consistent readings of this notoriously fickle parameter, we have optimized an established liposome-based assay for PH+/OH\\u2212 developed by Paula et\\u00a0al. that tracks the fluorescence decay of an encapsulated pH-sensitive dye in response to a step change in the external pH (19).; To make this study possible, we synthesized a series of\\u00a0extremophile-inspired lipids composed of two glycerophosphocholine heads connected by a membrane-spanning n-alkyl tether with each head ether-bonded to a phytanyl chain (T28, T32, and T36 (Fig.\\u00a01, A\\u2013C)), as well as several\\u00a0untethered lipids (U16, 1,2-di-O-phytanyl-sn-glycero-3-phosphocholine (Di-O-PhyPC), 1,2-diphytanoyl-sn-glycero-3-phosphocholine (DiPhyPC), and POPC (Fig.\\u00a01, D\\u2013G)) to seek answers to our questions about structure-permeability relationships of liposomes from pure lipid preparations at room temperature: First, how does the presence of a single tether between two lipid headgroups affect PH+/OH\\u2212 compared to the absence of a tether? Second, how does the length of the tether affect H+/OH\\u2212 permeability when the lengths of the untethered phytanyl chains are kept constant at 16 carbons? And third, how does the PH+/OH\\u2212 of an untethered lipid with a repeating isoprenoid methyl motif on one of its tails compare to that of a lipid with the methyl groups on both tails?; where [H+]f,o is the exterior proton concentration during the main decay phase, obtained from the pH measured after nigericin is added; A1 (mol\\u00a0m\\u22123), A2 (unitless), x1 (mol m\\u22123), x2 (mol m\\u22123), y1 (mol m\\u22123), and y2 (unitless) are the empirical fitting parameters from the calibration fits (Fig.\\u00a02, B and C, explained in Calculation); C (mol m\\u22123) is a constant of integration; PH+/OH\\u2212 (m s\\u22121) is the H+/OH\\u2212 permeability; r (m) is the most probable liposome radius as determined by DLS; and t (s) is time.;  The parameters C and PH+/OH\\u2212 are allowed to vary during fitting.;  The permeability values we obtained for membranes from POPC and DiPhyPC lipids (Table 1) are in reasonable agreement with some literature values reported for POPC, PH+/OH\\u2212\\u00a0= 7\\u00a0\\u00d7 10\\u22125 cm/s (23), and for DiPhyPC, PH+/OH\\u2212\\u00a0= 4.0\\u00a0\\u00d7 10\\u22126 cm/s (40) and 7.0\\u00a0\\u00d7 10\\u22126 cm/s (41).; Except in the case of bending rigidity and PH+/OH\\u2212 (last two columns), values are expressed as the mean \\u00b1 SE.;  Our goal was to find a range in which liposomes would exhibit PH+/OH\\u2212 values higher than in the absence of valinomycin, indicating that the buildup of a membrane potential accompanying H+ diffusion was being successfully prevented via K+ diffusion, but in which PH+/OH\\u2212 was independent of the valinomycin concentration, since a positive dependence would indicate protonophoric activity.;  As valinomycin is poorly soluble in water, we added it to liposome solutions via concentrated aliquots in DMSO; the resulting plot of PH+/OH\\u2212 versus valinomycin concentration is shown in Fig.\\u00a03.;  We chose DMSO as a solvent rather than ethanol (used in Paula et\\u00a0al. (19)), because DMSO has a smaller effect on PH+/OH\\u2212 (Fig.\\u00a03, leftmost data points).;  The H+/OH\\u2212 permeability increased from the baseline value after the addition of just 10 fM valinomycin and plateaued with increasing valinomycin concentrations until a point of inflection was reached around 100\\u00a0nM, at which point PH+/OH\\u2212 began to increase dramatically in the presence of more valinomycin.; Valinomycin concentrations up to 10\\u22127 M prevent membrane potential buildup without causing an increase in PH+/OH\\u2212. Before the axis break, points represent PH+/OH\\u2212 after the addition of the solvent vehicle only as indicated by the symbols: red squares, 2.66 \\u03bcL DMSO; brown circle, 1.33 \\u03bcL ethanol; orange diamond, 2.66 \\u03bcL ethanol; pink triangles, no solvent.; We explored the effect of transmembrane tethering, tether length, and the presence of isoprenoid methyl groups on one or both lipid tails on the PH+/OH\\u2212 values through liposomes formed from a series of synthetic lipids with small systematic structural variations.; Box-and-whisker plots showing PH+/OH\\u2212 values gathered from seven different lipids.; The PH+/OH\\u2212 permeability results from the lipids T28, T32, and T36 indicate that liposome membranes become more permeable to protons as tether length increases if the length of the untethered acyl chain is kept constant (Table 1).;  A Tukey honest significant difference test comparing the log-transformed PH+/OH\\u2212 values of these three lipids found the mean H+/OH\\u2212 permeability of liposomes composed of T28 lipids to be significantly smaller (p\\u00a0= 0.01) than that of T36 liposomes.;  As in prior research studies (19), we used log-transformed values for all statistical comparisons, because the populations of measured PH+/OH\\u2212 values failed tests of homogeneity of variance (Levene\\u2019s, p\\u00a0= 0.002; Brown-Forsythe, p\\u00a0= 0.02) unless the values were log-transformed (Levene\\u2019s, p\\u00a0= 0.72; Brown-Forsythe, p\\u00a0= 0.75).;  In the tethered lipids, the two untethered phytanyl chains have a combined length of 32 methylene groups\\u2014four carbons longer than the T28 tether and four carbons shorter than the T36 tether; the differences in PH+/OH\\u2212 may be attributable to increased hydrocarbon crowding in T28 and the presence of a \\u201cgap\\u201d in T36.;  Quantifying the entry of water molecules into the hydrophobic core in all six simulated membranes (including three untethered lipids, Table 2) revealed that log PH+/OH\\u2212 correlated with the total number of water molecules that spontaneously entered the membrane over 50\\u00a0ns (Pearson coefficient, r\\u00a0= 0.86; p\\u00a0= 0.027).;  A better correlation was found between log PH+/OH\\u2212 and the number of observations with at least three water molecules residing anywhere in the\\u00a0hydrophobic core of the membrane (Pearson coefficient, r\\u00a0= 0.99; p\\u00a0= 0.0003; Fig.\\u00a05\\nA).; Water within the hydrophobic region is predictive of log PH+/OH\\u2212. (A) Correlation plot of log PH+/OH\\u2212 as a function of the number of times at least three water molecules simultaneously occupied the hydrophobic region anywhere in the lipid membrane over the course of a 50\\u00a0ns MD simulation.;  The solid line is the best fit to a straight line; dotted lines show the 95% confidence interval from the fit. (B) Correlation plot of log PH+/OH\\u2212 as a function of the number of times a water cluster appeared in the hydrophobic region of the lipid membrane over the course of a 50\\u00a0ns MD simulation.;  To observe the effects of tethering, we compared PH+/OH\\u2212 values from membranes composed of U16 and T32, since T32\\u2019s structure is identical to two U16 lipids bonded together at the terminal carbons of their unbranched chains (Fig.\\u00a01, B and D).;  A two-tailed Student\\u2019s t-test comparing the log-transformed permeability values of T32 and U16 vesicles (Table 1) showed that they were not significantly different (p\\u00a0= 0.46); the spread of the U16 PH+/OH\\u2212 values fell entirely within the spread of the T32 PH+/OH\\u2212 values.;  Komatsu and Chong found that the H+/OH\\u2212 permeability of doubly tethered lipids isolated from Sulfolobus acidocaldarius was relatively invariant with respect to temperature from 25 to 80\\u00b0C, whereas PH+/OH\\u2212 values from standard (untethered) eggPC lipids increased dramatically with increasing temperature.;  The PH+/OH\\u2212 values from the two membrane compositions were comparable at room temperature but diverged greatly at higher temperatures (63).; Based on a two-tailed Student\\u2019s t-test comparing log-transformed PH+/OH\\u2212 values, the mean permeability of liposomes from Di-O-PhyPC lipids was lower (p\\u00a0= 0.059) than that from U16 lipids (Table 1).;  This variability, seen both in this work and in the literature (8, 16, 21, 25, 37, 41, 64, 65), represents a limitation of the usefulness of PH+/OH\\u2212 data (24, 26).;  We have attempted to identify and minimize potential sources of variation inherent in the process of gathering PH+/OH\\u2212 values, and we present a detailed discussion on this topic in Section S2 of the Supporting Material.;  The lipid T32, which was tethered by a straight 32-carbon chain, produced membranes\\u00a0with PH+/OH\\u2212 values similar to those of membranes composed of its untethered equivalent, U16, revealing that tethering does not affect PH+/OH\\u2212 at room temperature.;  However, PH+/OH\\u2212 values among tethered lipid membranes increased with increasing tether length if the number of carbons in the untethered acyl tail was constant.\",\n          \" In situ daily photosynthetic oxygen production ranged from 17.3 to 22.2 g O2 m-2 d-1, consistent with experimental production / irradiance measurements and modeling.;  Finally, the associated metabolism (photosynthetic oxygen production) and community metabolic potential (use of combined carbon sources) were evaluated.;  Dissolved oxygen sensors (optode technology, PONSEL Digisens) were suspended in the upper water column at depths of 15 cm and 50 cm, and dissolved oxygen saturation levels were recorded every 5 min.;  This method is considered to give a robust estimate of oxygen metabolism, as it takes account of physical diffusion.;  The dissolved oxygen was supersaturated in the upper layer of the lake most of the time and, therefore, some was lost to the atmosphere.; The microbial taxa were isolated and cultured to select eubacterial strains, both photosynthetic (cyanobacteria) and heterotrophic, using a single water sample collected in October 2011 at a depth of 0.5 m in the middle of the oxygenated layer.; Photosynthetic oxygen evolution (PSOE) was monitored as a proxy for photosynthetic activity. 30 mL samples were distributed in gas-tight polycarbonate flasks.;  The dissolved oxygen concentration was measured every twenty minutes over the 4\\u20136 h incubation period, the linearity of the oxygen concentration changes over time was checked, and the photosynthetic oxygen evolution (PSOE) rates were plotted against irradiance using the model of Platt et al. [48]:\\n; The PSOE parameters (\\u03b1, \\u03b2 and Pmax) obtained from the incubation experiments were used to model the instantaneous gross oxygen production rates at given times depending on the instantaneous light field and the actual chlorophyll a biomass.;  The rationale is described in [49], and the cumulative instantaneous oxygen production rates were used estimate the daily gross O2 production per square meter of lake (GPP):\\n; Assuming a constant respiration rate, calculated from samples incubated in the dark, the net oxygen production was obtained by subtracting the daily respiration from the daily gross production.; (A) Temperature (\\u00b0C, black diamonds), salinity (psu, blue squares) and dissolved oxygen concentration (mg L-1, red circles) are given for CLB station of Dziani Dzaha Lake in 2010 (closed symbols) and 2011 (open symbols). (B) Box-plot of temperature record at four selected depths during the two surveys (n = 2400). (C) Box-plot of all conductivity records during the two surveys (n = 370 for 2010; n = 252 for 2011).;  The subsurface diurnal cycles recorded during the 2011 survey showed a clear day/night pattern, but without total nocturnal depletion of the dissolved oxygen.;  At 31\\u00b0C and 52 psu, the oxygen saturation corresponded to a concentration of 5.6 mg O2 L-1 in the top 50 cm layer, and the dissolved oxygen never dropped far below this value even at night time.; The oxygen variation in the lake, measured at depths of 15 cm and 50 cm in 2011, showed a clear pattern of net production during the daytime and net consumption at night (S1 Fig).;  Salinity remained almost constant (51.3\\u201352.7 psu) during the two days of dissolved oxygen measurement.; Staehr et al. [23] method was used to fit optode in situ measurements, whereas Bright and Walsby [49] method was used to model production integral according to laboratory photosynthetic oxygen evolution (PSOE) measurements.;  The mean values for maximum oxygen production rates (Pmax) ranged from 5.61 to 11.76 mg O2 mg chl a-1 h-1 (equal to 175\\u2013368 \\u03bcmol O2 mg chl a-1 h-1) during the 2011 survey and the PSOE half-saturation flux density was 99 \\u03bcmol photons m-2 s-1.;  Both Pmax and \\u03b1 values, calculated from the oxygen production measurements, are within the standard range reported for cyanobacterial blooms.; \\u00a7 productivity is expressed as \\u03bcmol electrons m-2 s-1 for PAM fluorometry, and as \\u03bcmol O2 mg chl a-1 h-1 for photosynthetic oxygen evolution.; Using \\u03b1, \\u03b2 and Pmax derived from PSOE curves, the daily oxygen production was calculated using the method described in [49]. Using the input values listed in Table 4 for PSOE, a constant [chl a] of 612 \\u03bcg L-1 over the whole well-lit layer (mean of 2011 values excluding outliers), the averaged light measurements at a depth of 0.3 m (mean of 10 days, n = 720 for daytime periods) and a PAR attenuation coefficient Kd = 6.79 m-1, the instantaneous O2 gross production was calculated for 10 min intervals at a 10 cm resolution, giving a GPP of 19.4 g O2 m-2 d-1 (Table 3).; Dziani Dzaha is a shallow lake, with no vertical stratification during dry season except for dissolved oxygen, and the water buffering characteristics explain the pH stability despite high primary production rates at the surface.;  For carbon production, assuming a photosynthetic quotient PQ (molar ratio of O2 evolved to CO2 fixed) equal to 1 [9], the two days of dissolved oxygen measurements were consistent with a daily carbon fixation of 8.3 g C m-2 d-1.;  However, oxygen production in Dziani Dzaha was less than the early estimates of [14] for the upper limits of photosynthetic productivity (43\\u201357 g C m-2 d-1) in soda lakes.;  It has already been shown that, as oxygen is depleted, microbial communities use carbohydrates in preference to carboxylic acids and amino acids [67]. Several other compounds such as glycogen and pyruvic acid were products or end-products of the anaerobic metabolism of glucose, here preferentially metabolized in samples AN10 and AN11.;  The primary production rates in Dziani Dzaha, calculated from the in situ dissolved oxygen evolution, are close to the upper limit predicted for inland waters [16] and appear to be limited by light availability and self-shading of biomass.; Central panel: oxygen concentration evolution at 0.15 m (grey circles) and 0.5 m (black circles) depth in Dziani Dzaha Lake.;  Upper panel: example of linearization of production periods, used to calculate photosynthetic oxygen production rates.;  Lower panel: example of linearization of consumption periods, used to calculate oxygen respiration rates.; The lake water had a high level of dissolved matter and high alkalinity (10.6\\u201314.5 g L-1 eq.;  The phytoplankton biomass was very high, with a total chlorophyll a concentration of 524 to 875 \\u03bcg chl a L-1 depending on the survey, homogeneously distributed from surface to bottom (4 m).;  Heterotrophic bacterioplankton was extremely abundant, with cell densities up to 1.5 108 cells mL-1 in the whole water column.;  Measurements were converted to mg O2 L-1 as described in [22], correcting for temperature and salinity.; \\nwhere [chl a] is the concentration of chlorophyll a in the ethanol extract in \\u03bcg mL-1 and A665 and A750 are the absorbance readings at 665 nm and 750 nm, respectively, for a 1 cm path optical glass cuvette.; Water samples were inoculated on Z8 medium [41] supplemented with salt (NaCl 20 g L-1), in liquid and solid medium (7 g L-1 of agar).; (A) Temperature (\\u00b0C, black diamonds), salinity (psu, blue squares) and dissolved oxygen concentration (mg L-1, red circles) are given for CLB station of Dziani Dzaha Lake in 2010 (closed symbols) and 2011 (open symbols). (B) Box-plot of temperature record at four selected depths during the two surveys (n = 2400). (C) Box-plot of all conductivity records during the two surveys (n = 370 for 2010; n = 252 for 2011).;  The alkalinity is high, with levels comprised between 10.6 and 14.5 g L-1 eq.; Dissolved anions, cations (mg L-1), and total alkalinity TAL (mg L-1 eq.; Ward\\u2019s method clustering based on the records of the six major dissolved elements (Na+, K+, Mg2+, Ca2+, SO42-, Cl-) expressed as mg L-1, and total alkalinity expressed as mg L-1 eq.CO32-.; The total chlorophyll a concentration in Dziani Dzaha was extremely high, mainly around 524 \\u03bcg chl a L-1 during the 2010 survey and 875 \\u03bcg chl a L-1 in 2011 (n = 74), with a uniform vertical distribution (Fig 4A).;  Without outliers, the mean values were 685\\u00b166 \\u03bcg chl a L-1 in 2010 and 702 \\u00b1 48 \\u03bcg chl a L-1 in 2011.;  There was only one exception on Sep. 29th, 2011, when the surface values dropped to 175 \\u03bcg chl a L-1 with the concentration reaching 2470 \\u03bcg chl a L-1 just above the sediment (outliers in Fig 4A and 4B).;  At 31\\u00b0C and 52 psu, the oxygen saturation corresponded to a concentration of 5.6 mg O2 L-1 in the top 50 cm layer, and the dissolved oxygen never dropped far below this value even at night time.;  Linear regression of dark records gave a respiration rate of between -0.93\\u00b10.024 mg O2 L-1 h-1 and -0.59\\u00b10.012 mg O2 L-1 h-1 depending on the depth and day considered (Table 3), whereas the apparent in situ net production during the first three hours of the day varied between 1.19\\u00b10.06 mg O2 L-1 h-1 and 3.59\\u00b10.2 mg O2 L-1 h-1.; Using \\u03b1, \\u03b2 and Pmax derived from PSOE curves, the daily oxygen production was calculated using the method described in [49]. Using the input values listed in Table 4 for PSOE, a constant [chl a] of 612 \\u03bcg L-1 over the whole well-lit layer (mean of 2011 values excluding outliers), the averaged light measurements at a depth of 0.3 m (mean of 10 days, n = 720 for daytime periods) and a PAR attenuation coefficient Kd = 6.79 m-1, the instantaneous O2 gross production was calculated for 10 min intervals at a 10 cm resolution, giving a GPP of 19.4 g O2 m-2 d-1 (Table 3).;  Assuming a constant mean respiration rate, for the whole period, equal to 0.6 mg O2 L-1 h-1 calculated from incubations in bottles in the dark, the resulting net production was 8 g O2 m-2 d-1 (Table 3).;  No significant differences were found in the total bacterial cell counts between days of year, years or sampling depths, with an average density over the whole survey of 1.38 x 108 cells mL-1 (SD = 22.4%, n = 77, Fig 7B).; Dissolved organic carbon concentrations were relatively homogeneous on the water column, with a mean of 77.5 mg L-1 (n = 11) in 2010 and 66,6 mg L-1 (n = 25) in 2011, and C:N ratio for dissolved organic matter enriched in carbon relative to particulate matter (Table 7).;  Assuming a mean total particulate carbon of 35 mg C L-1 (Table 5) and a total photosynthetic carbon biomass of 140 g C m-2 for a 4 m deep water column, a daily production of 8.3 g C m-2 d-1 represented only 6% of the total stock.;  A mean abundance of 1.38 x 108 cells mL-1 is much higher than in other aquatic ecosystems (estuaries, lakes and oceans), where values generally range from 1.4 105 to 1.6 107 cells mL-1 [63]. Flow cytometry did not clearly discriminated bacterial populations during the two surveys, except for the mean ratio of high to low DNA cells that differed in 2010 and 2011 by a factor of three.\",\n          \"Berkeley Pit Lake, Butte, Montana is a 540 m. deep abandoned open-pit copper mine filled with over 140 billion liters of acidic, metal-sulfate contaminated water.;  The Pit itself is over 540 m deep with a surface area of 3.2 km2 and is continually filling with metal-sulfate rich, acidic water (pH 2.5), at a rate of 10 million L/day.;  This represents roughly 140 billion L. of contaminated water and constitutes an important component of the largest EPA Superfund site in the United States.1; In 1995 we began to study the microbes inhabiting the waters of this Pit Lake as if they were inhabitants of a new and exotic ecosystem.;  Over the past fifteen years we have studied the secondary metabolism of several microbes isolated from the water and sediments of this ecosystem under a variety of physicochemical conditions to determine whether or not they produce metabolites with desirable bioactivity.;  Experiments were conducted in 96-well plates for 24 h in 37 \\u00b0C water-jacketed CO2 incubators (ThermoForma).; The compounds were isolated either for their inhibition of the signal transduction enzyme caspase-1 or because of their structural similarity to these inhibitors.;  We routinely target the enzymes matrix metalloproteinase-3 (MMP-3), caspase-1 and caspase-3.;  The lead compounds presented in this manuscript were isolated because of their ability to inhibit caspase-1.; Several researchers have demonstrated that caspase-1 inhibitors have shown promise in delaying the onset of Huntington's disease10 and amyotropic lateral sclerosis,11 and in mitigating the effects of stroke12 and multiple sclerosis.13, 14 All of these diseases exhibit autoimmune phenomena.; The up-regulation of caspase-1 and concomitant chronic inflammation has been associated with a number of different pathologies including the development of insulin resistance in obesity related diabetes,16 degeneration of retinal capillaries associated with diabetes and galactosemia,17 the demyelination of neurons in multiple sclerosis,11,18 and the formation of amyloid plaques in Alzheimer's disease.19 High levels of caspase-1 and interleukin-1\\u03b2 have been found in certain cancers20 by many different researchers: acute myelogenous leukemia,21 melanoma,22\\u201323 certain glioblastomas24\\u201325 and pancreatic cancers,26\\u201329 certain breast cancers,30 and human cancer xenografts31 all of which may be exacerbated by chronic inflammation associated with activation of the inflammasome.; Caspase-1 is also down-regulated in many solid tumor cancers and activation of caspase-1 in prostate cancer and ovarian cancer may be required for apoptotic breakdown of tumors.;  The development of new caspase-1 inhibitors will not only provide potential chemotherapeutics but will also provide tools for the investigation of the intricacies of signal transduction.;  This organic extract was active in all three enzyme inhibition assays, although in this study we focused on the caspase-1 inhibitors.; Compounds 1, 2, 4, and 6\\u20139 were evaluated for their ability to inhibit caspase-1 in vitro, and the most active compounds and closely related analogues were evaluated for their ability to inhibit the production of interleukin 1-\\u03b2 in THP-1 cells (pro-monocytic leukemia cell line).;  Both berkazaphilone B (2) and C (4) had IC100 values of 25 \\u03bcM against caspase-1, while berkazaphilone A (1), penisimplicissin (8), and compound 9 were completely inhibitory at a concentration of 250 \\u03bcM.;  It was interesting to note that in several microarray experiments caspase-1 was upregulated almost exclusively in different leukemia cell lines.53\\u201356;  This extract demonstrated inhibition of caspase-1 and MMP-3, antimicrobial activity against Staphylococcus aureus and Escherichia coli, and brine shrimp lethality.;  The resulting cell pellet was suspended at 1.0 \\u00d7 106 cells/mL and exposed to caspase-1 inhibitors at concentrations described below (0.5%\\u20130.005%), LPS [20ng/mL] and TiO2 nanowires (100 \\u03bcg/mL).\"\n        ],\n        \"semantic_type\": \"\",\n        \"description\": \"\"\n      }\n    }\n  ]\n}"
            }
          },
          "metadata": {},
          "execution_count": 2
        }
      ],
      "source": [
        "merged_df['Sentences'] = merged_df['Sentences_x'] + ';' + merged_df['Sentences_y']\n",
        "merged_df.drop(columns=['Sentences_x', 'Sentences_y'], inplace=True)\n",
        "merged_df = merged_df.drop_duplicates(subset=['Sentences'], keep='first')\n",
        "\n",
        "merged_df.to_csv(\"gene_chemicals2.csv\")\n",
        "merged_df\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "f7ded484",
      "metadata": {
        "id": "f7ded484"
      },
      "outputs": [],
      "source": [
        "import pandas as pd\n",
        "\n",
        "def remove_duplicate_sentences(text):\n",
        "    if pd.isna(text):\n",
        "        return text\n",
        "\n",
        "    # Split the text into sentences\n",
        "    text = text.lower()\n",
        "    sentences = text.split(\";\")\n",
        "\n",
        "    # Remove duplicate sentences\n",
        "    unique_sentences = list(set(sentences))\n",
        "    unique_sentences = [s.strip() for s in unique_sentences if s.strip() != \"\"]\n",
        "\n",
        "    # Sort the unique sentences alphabetically\n",
        "    unique_sentences.sort()\n",
        "\n",
        "    # Join the unique sentences back into a single string\n",
        "    result = \";\".join(unique_sentences)\n",
        "\n",
        "    return result\n",
        "\n",
        "# Read the CSV file\n",
        "df = pd.read_csv(\"gene_chemicals2.csv\")\n",
        "\n",
        "# Apply the remove_duplicate_sentences function to each value in the \"Sentences\" column\n",
        "df[\"Sentences\"] = df[\"Sentences\"].apply(remove_duplicate_sentences)\n",
        "\n",
        "# Save the modified DataFrame to a new CSV file\n",
        "df.to_csv(\"gene_chemicals_up2.csv\", index=False)\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "494e77a0",
      "metadata": {
        "id": "494e77a0"
      },
      "outputs": [],
      "source": [
        "import pandas as pd\n",
        "\n",
        "# Read the CSV file\n",
        "df = pd.read_csv(\"gene_chemicals_up2.csv\")\n",
        "\n",
        "# Function to filter lines containing both chemicals_x and chemicals_y terms\n",
        "def filter_lines(sentences, chemicals_x, chemicals_y):\n",
        "    if pd.isna(sentences):\n",
        "        return \"\"\n",
        "\n",
        "    filtered_lines = []\n",
        "    for line in sentences.split(';'):\n",
        "        if chemicals_x.lower() in line.lower() and chemicals_y.lower() in line.lower():\n",
        "            filtered_lines.append(line)\n",
        "    return ';'.join(filtered_lines)\n",
        "\n",
        "# Apply the filter_lines function to each row\n",
        "df['co-occur_sentences'] = df.apply(lambda row: filter_lines(row['Sentences'], row['Chemicals'], row['Gene']), axis=1)\n",
        "\n",
        "# Remove rows with empty 'co-occur_sentences'\n",
        "df = df[df['co-occur_sentences'].str.strip() != \"\"]\n",
        "\n",
        "# Save the modified DataFrame to a new CSV file\n",
        "df.to_csv(\"Gene_Chemicals_occur_sentences.csv\", index=False)\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "9b17118a",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "9b17118a",
        "outputId": "c7570f45-9c83-4ae5-b5b6-68cd4ec87e72"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "        Chemicals ChemicalsID      PMID   PMCID_x         Gene  GeneID  \\\n",
            "0           water     D014867  32533304   7347518          tap   10482   \n",
            "1            nadh     D009243  37577439  10416648        ndh-2    1660   \n",
            "2          saline     D012965  34256694   8278772         pip1  856754   \n",
            "3           water     D014867  34256694   8278772         pip1  856754   \n",
            "4           lipid     D008055  27276261   4906265           ph    5053   \n",
            "5       hydroxide     C031356  27276261   4906265           ph    5053   \n",
            "6   phospholipids     D010743  27276261   4906265           ph    5053   \n",
            "7           lipid     D008055  27276261   4906265           ph    5053   \n",
            "8     valinomycin     D014634  27276261   4906265           ph    5053   \n",
            "9         carbons     D002244  27276261   4906265           ph    5053   \n",
            "10         lipids     D008055  27276261   4906265           ph    5053   \n",
            "11     isoprenoid     D013729  27276261   4906265           ph    5053   \n",
            "12          water     D014867  27276261   4906265           ph    5053   \n",
            "13          water     D014867  28045976   5207672          l-1   16728   \n",
            "14          water     D014867  28045976   5207672         ml-1   16728   \n",
            "15         oxygen     D010100  28045976   5207672          l-1   16728   \n",
            "16          water     D014867  24927538   4156692          fog  161882   \n",
            "17       nitrogen     D009584  24927538   4156692          fog  161882   \n",
            "18             o2     D010100  24927538   4156692          fog  161882   \n",
            "19            eap     C005448  30400922   6220464  osteocalcin   12097   \n",
            "20            res  D000068296  30400922   6220464  osteocalcin   12097   \n",
            "\n",
            "                                            Sentences  \\\n",
            "0   a great variety of fungi was detected in drink...   \n",
            "1   besides the expected aerobic etc, somewhat sur...   \n",
            "2   0.5 mmol/l nahco3 plus 99.5 mmol/l nacl, 0.75 ...   \n",
            "3   0.5 mmol/l nahco3 plus 99.5 mmol/l nacl, 0.75 ...   \n",
            "4   a better correlation was found between log ph+...   \n",
            "5   a better correlation was found between log ph+...   \n",
            "6   a better correlation was found between log ph+...   \n",
            "7   a better correlation was found between log ph+...   \n",
            "8   a better correlation was found between log ph+...   \n",
            "9   a better correlation was found between log ph+...   \n",
            "10  a better correlation was found between log ph+...   \n",
            "11  a better correlation was found between log ph+...   \n",
            "12  a better correlation was found between log ph+...   \n",
            "13  (a) temperature (°c, black diamonds), salinity...   \n",
            "14  a chemical analysis was undertaken to determin...   \n",
            "15  (a) temperature (°c, black diamonds), salinity...   \n",
            "16  a key question for life on exoplanets is how l...   \n",
            "17  after carbon, nitrogen is arguably the most im...   \n",
            "18  a variety of energetic processes such as auror...   \n",
            "19  (eap:tm): mixed formula consisted of eap and t...   \n",
            "20  (eap:tm): mixed formula consisted of eap and t...   \n",
            "\n",
            "                                   co-occur_sentences interaction_type  \n",
            "0   according to the who, algae producing toxins, ...           others  \n",
            "1   the integral membrane protein nadh dehydrogena...           others  \n",
            "2   5 could increase sensitivity to saline-alkalin...           others  \n",
            "3   5 in c. rosea seedlings caused by stress facto...           others  \n",
            "4   a fluorescence-based permeability assay was op...           others  \n",
            "5   this work explores the proton/hydroxide permea...           others  \n",
            "6   these lipids usually contain ether bonds conne...           others  \n",
            "7   a fluorescence-based permeability assay was op...           others  \n",
            "8   all permeability assays were performed in the ...           others  \n",
            "9   among tethered lipid monolayers, ph+/oh− incre...           others  \n",
            "10  a tukey honest significant difference test com...           others  \n",
            "11  a fluorescence-based permeability assay was op...           others  \n",
            "12  a better correlation was found between log ph+...           others  \n",
            "13  assuming a mean total particulate carbon of 35...           others  \n",
            "14  heterotrophic bacterioplankton was extremely a...           others  \n",
            "15  (a) temperature (°c, black diamonds), salinity...           others  \n",
            "16  based on our understanding of life on earth we...           others  \n",
            "17  based on our understanding of life on earth we...           others  \n",
            "18  based on our understanding of life on earth we...           others  \n",
            "19  however, all eap:tm single or mixed formulatio...           others  \n",
            "20  despite these issues, serum balp and osteocalc...           others  \n",
            "DataFrame saved to output_co-occurrence_chemicals_gene.csv\n"
          ]
        }
      ],
      "source": [
        "import pandas as pd\n",
        "import re\n",
        "\n",
        "# Read the CSV file with specified data types for each column\n",
        "#dtype_dict = {'Chemicals_x': 'str', 'ChemicalsID_x': 'str','PMID': 'str','PMCID_x':'str','Chemicals_y': 'str', 'ChemicalsID_y': 'str','Sentences':'str','co-occur_sentences':'str'}  # Specify the data types for each column\n",
        "merged_df = pd.read_csv('Gene_Chemicals_occur_sentences.csv')\n",
        "\n",
        "selected_columns = ['Chemicals', 'ChemicalsID', 'PMID', 'PMCID_x', 'Gene', 'GeneID', 'Sentences', 'co-occur_sentences']\n",
        "merged_df = merged_df[selected_columns]\n",
        "\n",
        "merged_df['co-occur_sentences'] = merged_df['co-occur_sentences'].str.replace('\\n', '')\n",
        "\n",
        "\n",
        "def extract_interaction_types(sentence):\n",
        "    if pd.isna(sentence) or sentence.strip() == '':\n",
        "        return \"NULL\"\n",
        "\n",
        "    interaction_types = ['Inhibition', 'Activation', 'Proliferation', 'Allosteric', 'Agonist', 'Antagonist']\n",
        "    for interaction_type in interaction_types:\n",
        "        if re.search(r'\\b' + re.escape(interaction_type.lower()) + r'\\b', sentence.lower(), re.IGNORECASE):\n",
        "            return interaction_type\n",
        "    return 'others'\n",
        "\n",
        "# Drop rows with NaN values in the 'co-occur_sentences' column\n",
        "merged_df.dropna(subset=['co-occur_sentences'], inplace=True)\n",
        "merged_df = merged_df.loc[:, ~merged_df.columns.str.contains('^Unnamed')]\n",
        "\n",
        "# Create a copy of the DataFrame\n",
        "merged_df_copy = merged_df.copy()\n",
        "\n",
        "# Apply the function to the 'co-occur_sentences' column and create a new column 'interaction_type'\n",
        "merged_df_copy['interaction_type'] = merged_df_copy['co-occur_sentences'].apply(extract_interaction_types)\n",
        "\n",
        "print(merged_df_copy)\n",
        "\n",
        "\n",
        "merged_df_copy.to_csv('output_co-occurrence_chemicals_gene.csv', index=False)\n",
        "\n",
        "print(\"DataFrame saved to output_co-occurrence_chemicals_gene.csv\")\n"
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "eOsBobMitVcc"
      },
      "id": "eOsBobMitVcc",
      "execution_count": null,
      "outputs": []
    }
  ],
  "metadata": {
    "kernelspec": {
      "display_name": "Python 3 (ipykernel)",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.11.4"
    },
    "colab": {
      "provenance": [],
      "toc_visible": true
    }
  },
  "nbformat": 4,
  "nbformat_minor": 5
}